### ISSUE TITLE
Build race between generating eventing headers and compiling source code needing them

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/coreclr#11682]
I just hit a build failure in my Yocto ARM build where CMake already tried to compile `dotnetruntime.cpp` before the `genXplatEventing.py` script had generated the required `clretwallmain.h` header:

```
[...]
| [  1%] Building CXX object src/inc/CMakeFiles/corguids.dir/__/pal/prebuilt/idl/gchost_i.cpp.o
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/bin/obj/Linux.arm.Release/Generated/eventpipe/dotnetruntime.cpp:12:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/vm/common.h:306:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/vm/eepolicy.h:15:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/vm/vars.hpp:70:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/vm/eeprofinterfaces.h:19:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/inc/profilepriv.h:128:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/inc/profilepriv.inl:18:
| In file included from /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/vm/eetoprofinterfaceimpl.h:23:
| /home/kai/sources/build/tmp/work/cortexa9hf-neon-target-linux-gnueabi/coreclr/2.0-25316-03-r0/git/src/inc/eventtracebase.h:308:10: fatal error: 'clretwallmain.h' file not found
| #include "clretwallmain.h"
|          ^
| [  1%] Building CXX object src/inc/CMakeFiles/corguids.dir/__/pal/prebuilt/idl/mscorsvc_i.cpp.o
|  Generating Dummy Event Headers
|  Generating Event Headers
| [  1%] Building CXX object src/inc/CMakeFiles/corguids.dir/__/pal/prebuilt/idl/tlbimpexp_i.cpp.o
| [  1%] Built target GeneratedEventingFiles
| [  1%] Building CXX object src/inc/CMakeFiles/corguids.dir/__/pal/prebuilt/idl/clrprivappxhosting_i.cpp.o
[...]
```

I just restarted the build (with the generated files already in place) and it succeeded

**Comment by janvorli** [comment]
The #12114 fixes this issue.

**Comment by Tragetaschen** [comment]
I think so, too.

**Comment by Tragetaschen** [comment]
I've just seen it again building the `v2.0.0-preview2` tag. It's while building a different file, though:
```
| [  1%] Building CXX object src/dlls/mscordac/CMakeFiles/mscordacobj.dir/mscordac.cpp.o
| In file included from [...]/bin/obj/Linux.arm.Release/Generated/eventpipe/dotnetruntime.cpp:12:
| In file included from [...]/src/vm/common.h:306:
| In file included from [...]/src/vm/eepolicy.h:15:
| In file included from [...]/src/vm/vars.hpp:70:
| In file included from [...]/src/vm/eeprofinterfaces.h:19:
| In file included from [...]/src/inc/profilepriv.h:128:
| In file included from [...]/src/inc/profilepriv.inl:18:
| In file included from [...]/src/vm/eetoprofinterfaceimpl.h:23:
| [...]/src/inc/eventtracebase.h:308:10: fatal error: 'clretwallmain.h' file not found
| #include "clretwallmain.h"
|          ^~~~~~~~~~~~~~~~~
[...]
|  Generating Event Headers
```

**Comment by RussKeldorph** [comment]
#12114 was not included in v2.0.0-preview2 as far as I can tell, so a break would be expected.  It does appear to be in v2.0.0.
